src/netbase.cpp

```diff
@@ -56,29 +57,64 @@ std::vector<CNetAddr> WrappedGetAddrInfo(const std::string& name, bool allow_loo
     // hostname lookups.
     ai_hint.ai_flags = allow_lookup ? AI_ADDRCONFIG : AI_NUMERICHOST;
 
+    std::set<CNetAddr> resolved_addresses_set;
+
     addrinfo* ai_res{nullptr};
-    const int n_err{getaddrinfo(name.c_str(), nullptr, &ai_hint, &ai_res)};
-    if (n_err != 0) {
-        return {};
-    }
```

Comment:
I removed the error handling here for two reasons:

Firstly it just returns an empty vector which is what I believe this function will do anyway if it fails to get address info.

Secondly, if `getaddrinfo` doesn't allow loopback only IPV6 then that is returned as an error code 1 "address family for hostname not supported" so this specific case would have to be allowed.

Reply:
Are you sure that `getaddrinfo()` would return error (ie `n_err != 0`) and still set something useful in `ai_res`? That sounds strange.

I am worried that it may return an error (for whatever reason) and set `ai_res` to a bogus pointer which we later try to dereference.

Just returning from the lambda and not from `WrappedGetAddrInfo()` if `getaddrinfo()` returns an error seems reasonable to me.

Reply:
It was working but I agree handling the error is safer.

Also I was wrong that it would need to handle the specific "address family for hostname not supported" error.

Implemented your suggestion of just returning from the lambda.